### PR TITLE
feat(sync): add a fallback mechanism when the pairing does't work

### LINKED ISSUE #15750: Implement pairing fallback
### Description

When syncing fails, we need a fallback for users.

That fallback is to let them use their seed phrase and then the pairing will be done later with a new API.
(design pending for the fallback button)

After logging in from the seed phrase a notification will be added to the Notification center. It will depend one which side you are on.

New device:
image
image

Old device:
image
image

Design link: https://www.figma.com/design/17fc13UBFvInrLgNUKJJg5/Kuba%E2%8E%9CDesktop?node-id=1023-298171&t=38l091rwjrozIeAN-0

Status-go branch: https://github.com/status-im/status-go/tree/feature/fallback-pairing-seed

### Acceptance Criteria

- When the syncing fails, a button to use the seed phrase is available
- When the user logs in with a seed phrase after failing the sync, both devices will receive a notification (see designs)
- When the pairing is accepted, the two devices should be paired. They will appear in the Devices list in settings and messages from that profile will appear to both (as normal when pairing)
- If the pairing is ignored, nothing special happens

### PR DESCRIPTION
**Pull request by jrainville**
### What does the PR do

Fixes #15750

When the pairing fails, the UI now let's the user use the seed phrase instead. When they do, a call is send to the original device and both instances will show an AC notif. When the original device accepts the pairing, the call is made to pair and sync the devices and the AC notifs get deleted

### Affected areas

Pairing and AC notifs

### Screenshot of functionality (including design for comparison)

- [x] I've checked the design and this PR matches it (I didn't add the Retry button because I don't know if we support it for now, going back and doing it again does the same)

[pairing.webm](https://github.com/user-attachments/assets/22965016-2e97-426f-baca-479b87665f60)

### Impact on end user

This should only be a positive, since if the pairing failed before, you were stuck. Now, the pairing can still happen even when there is a failure.

### How to test

I guess you could set both devices on different networks, that way you know it will fail.

I personally tested by faking a failure in status-go.

I put `err = errors.New("FAKE ERROR")` in `status-go/server/pairing/client.go`  on line 99.

### Risk 

Maybe the normal flow of signing in with a seed phrase could be impacted, but it shouldn't as we won't have an installation ID, so it should just do nothing.

Tick **one**:
- [x] Low risk: 2 devs MUST perform testing as specified above and attach their results as comments to this PR **before** merging.
- [ ] High risk: QA team MUST perform additional testing in the specified affected areas **before** merging.

### CODE REVIEW

**Comment by jrainville** [comment]
I tested only Desktop -> Desktop, but Mobile -> Desktop and Desktop -> Mobile should work too.
The mobile PR is here: https://github.com/status-im/status-mobile/pull/20766

**Comment by jrainville** [comment]
Status-go PR : https://github.com/status-im/status-go/pull/5794

File: ui/app/mainui/activitycenter/views/ActivityNotificationNewDevice.qml [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+            }
+        }
+    ]
+
+    Component {
+        id: ctaFlatBtnComponent
+
+        StatusFlatButton {
+            size: StatusBaseButton.Size.Small
+            text: d.ctaText
```
Comment (caybro) [inline ui/app/mainui/activitycenter/views/ActivityNotificationNewDevice.qml:143]: Same here, the `ctaText` is identical, so put the whole component directly under `d` with the same text

File: ui/app/mainui/activitycenter/views/ActivityNotificationNewDevice.qml [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+import StatusQ.Core.Theme 0.1
+import StatusQ.Components 0.1
+
+import shared 1.0
+import shared.panels 1.0
+import utils 1.0
+
+import "../panels"
+import "../popups"
+import "../stores"
```
Comment (caybro) [inline ui/app/mainui/activitycenter/views/ActivityNotificationNewDevice.qml:15]: No relative imports pls

File: ui/app/mainui/activitycenter/popups/ActivityCenterPopup.qml [hunk 3]
```diff
@@ -369,4 +402,88 @@ Popup {
             onCloseActivityCenter: root.close()
         }
     }
+
+    Component {
+        id: pairDeviceDialog
+
+        StatusDialog {
```
Comment (caybro) [inline ui/app/mainui/activitycenter/popups/ActivityCenterPopup.qml:409]: When does this get destroyed? You should either set `destroyOnClose: true` or `destroy()` it manually in the `onClosed` handler

File: ui/app/mainui/activitycenter/popups/ActivityCenterPopup.qml [hunk 4]
```diff
@@ -369,4 +402,88 @@ Popup {
...
+            width: 620
+            closePolicy: Popup.CloseOnPressOutside
+
+            title: qsTr("Pair new device and sync profile")
+
+            contentItem: ColumnLayout {
+                spacing: 16
+                StatusBaseText {
+                    Layout.fillWidth: true
+                    font.pixelSize: 15
```
Comment (caybro) [inline ui/app/mainui/activitycenter/popups/ActivityCenterPopup.qml:422]: No need, default

File: ui/app/mainui/activitycenter/popups/ActivityCenterPopup.qml [hunk 5]
```diff
@@ -369,4 +402,88 @@ Popup {
...
+                    font.pixelSize: 27
+                    font.weight: Font.Medium
+                    text: deviceId
+                }
+            }
+
+            footer: StatusDialogFooter {
+                leftButtons: ObjectModel {
+                   StatusFlatButton {
+                        text: "Cancel"
```
Comment (caybro) [inline ui/app/mainui/activitycenter/popups/ActivityCenterPopup.qml:438]: `qsTr()` missing

File: ui/app/mainui/activitycenter/popups/ActivityCenterPopup.qml [hunk 6]
```diff
@@ -369,4 +402,88 @@ Popup {
...
+                    Layout.alignment: Qt.AlignHCenter
+                    font.pixelSize: 27
+                    font.weight: Font.Medium
+                    text: deviceId
+                }
+                Item {
+                    Layout.fillWidth: true
+                }
+            }
+            footer: null
```
Comment (caybro) [inline ui/app/mainui/activitycenter/popups/ActivityCenterPopup.qml:486]: Does this really have no buttons? :thinking:

File: ui/app/mainui/activitycenter/popups/ActivityCenterPopup.qml [hunk 7]
```diff
@@ -369,4 +402,88 @@ Popup {
...
+            width: 620
+            closePolicy: Popup.CloseOnPressOutside
+
+            title: qsTr("Pair this device and sync profile")
+
+            contentItem: ColumnLayout {
+                spacing: 16
+                StatusBaseText {
+                    Layout.fillWidth: true
+                    font.pixelSize: 15
```
Comment (caybro) [inline ui/app/mainui/activitycenter/popups/ActivityCenterPopup.qml:472]: Default

File: ui/app/mainui/activitycenter/popups/ActivityCenterPopup.qml [hunk 8]
```diff
@@ -369,4 +402,88 @@ Popup {
...
+                    }
+                }
+            }
+        }
+    }
+
+    Component {
+        id: checkOtherDeviceDialog
+
+        StatusDialog {
```
Comment (caybro) [inline ui/app/mainui/activitycenter/popups/ActivityCenterPopup.qml:460]: Same remark about the destruction

File: ui/app/mainui/activitycenter/popups/ActivityCenterPopup.qml (hunk 6 above)
Comment (jrainville) [reply ui/app/mainui/activitycenter/popups/ActivityCenterPopup.qml:486]: Yup https://www.figma.com/design/17fc13UBFvInrLgNUKJJg5/Kuba%E2%8E%9CDesktop?node-id=40765-401872&node-type=frame&t=dGj7XVHOMQlL3Juw-0

File: ui/app/mainui/activitycenter/popups/ActivityCenterPopup.qml [hunk 9]
```diff
@@ -369,4 +402,94 @@ Popup {
             onCloseActivityCenter: root.close()
         }
     }
+
+    function trucateDeviceId(deviceId) {
```
Comment (caybro) [inline ui/app/mainui/activitycenter/popups/ActivityCenterPopup.qml:406]:
```suggestion
    function truncateDeviceId(deviceId) {
```

File: ui/app/mainui/activitycenter/views/ActivityNotificationNewDevice.qml [hunk 10]
```diff
@@ -0,0 +1,131 @@
...
+                        onClicked: {
+                            root.moreDetailsClicked()
+                        }
+                    }
+                }
+            }
+        }
+    }
+
+    ctaComponent: undefined
```
Comment (caybro) [inline ui/app/mainui/activitycenter/views/ActivityNotificationNewDevice.qml:111]: Shouldn't this `ctaComponent` be the `StatusFlatButton` above?

**Review by caybro — COMMENTED** [review]
Looks good now, just 2 minor things

**Review by caybro — APPROVED** [review]
LGTM

File: src/app_service/service/activity_center/service.nim [hunk 11]
```diff
@@ -88,7 +88,7 @@ QtObject:
     result.chatService = chatService
 
   proc handleNewNotificationsLoaded(self: Service, activityCenterNotifications: seq[ActivityCenterNotificationDto]) =
-    # For now status-go notify about every notification update regardless active group so we need filter manulay on the desktop side
+    # For now status-go notify about every notification update regardless active group so we need filter manualy on the desktop side
```
Comment (igor-sirotin) [inline src/app_service/service/activity_center/service.nim:91]: 😂 
```suggestion
    # For now status-go notify about every notification update regardless active group so we need filter manually on the desktop side
```

**Review by igor-sirotin — APPROVED** [review]
👍

**Comment by jrainville** [comment]
I can reproduce the issue. I happens when the device is completely cut from the internet. It doesn't generate the AC notification for itself correctly, though it still sends the one for the other device

**Comment by jrainville** [comment]
@glitchminer nice find.

The problem was that when you have no internet, the profile is not fetched at all, so you need to create a new profile and it's a different flow and function called in the module. So I added the call to put the AC notification in that new place in the module and everything works as expected.

You can resume testing with the new build.

**Comment by jrainville** [comment]
@glitchminer Fixed!

image

**Comment by jrainville** [comment]
> regression - sync codes are different again and pair button is shown on both instances

I can confirm the regression. Both instances have the buttons.

Not sure how that happened, I only changed the UI last time :thinking:

**Comment by jrainville** [comment]
@glitchminer alright, it's fixed and I tested the flow multiple times now and it works well. I'm hoping everything is fine already :smile:

**Pull request merged**